Userland/Libraries/LibC/sys/types.h

```diff
@@ -57,16 +57,21 @@ typedef __WINT_TYPE__ wint_t;
 typedef uint32_t ino_t;
 typedef int64_t off_t;
 
+typedef uint32_t blksize_t;
+typedef uint32_t blkcnt_t;
 typedef uint32_t dev_t;
 typedef uint16_t mode_t;
 typedef uint32_t nlink_t;
-typedef uint32_t blksize_t;
-typedef uint32_t blkcnt_t;
+
 typedef int64_t time_t;
 typedef uint32_t useconds_t;
+typedef uint16_t u_mode_t;
 typedef int32_t suseconds_t;
 typedef uint32_t clock_t;
 
+typedef int64_t fsblkcnt_t;
+typedef int64_t fsfilcnt_t;
```

Comment:
These are `uint64_t` on other systems

Reply:
Wait, really? I was studying a few man pages which seemed a bit unclear about this and I wondered what happened in very specific exceptional cases, having remembered a few misfortunes that I dealt with when I struggled with using `dd` right under Linux... where the result to how much free space there could've been negative.

I'm assuming what I saw was the result of subtraction, since the available blocks back then  was `0`.

Reply:
https://github.com/openbsd/src/blob/2207c4325726fdc5c4bcd0011af0fdf7d3dab137/sys/sys/types.h#L151-L152
https://github.com/openbsd/src/blob/2207c4325726fdc5c4bcd0011af0fdf7d3dab137/sys/sys/_types.h#L46-L47

Reply:
OHHHHHH, Yeah, I had no idea where the hell `__fsfilcnt_t` was and was thrown off by the second definition in the file.

TIL: [FreeBSD does ` sys/sys/_types.h `](https://github.com/openbsd/src/blob/2207c4325726fdc5c4bcd0011af0fdf7d3dab137/sys/sys/_types.h).